typing/typecore.ml

```diff
@@ -3807,8 +3807,9 @@ and type_expect_
         exp_attributes = sexp.pexp_attributes;
         exp_env = env }
   | Pexp_constraint (sarg, sty) ->
-      let (ty', exp_extra) = type_constraint env sty in
-      let arg = type_argument env sarg ty' (instance ty') in
+      let (ty, exp_extra) = type_constraint env sty in
+      let ty' = instance ty in
```

Comment:
Could write the `instance tty` directly in `exp_type` field below. But it's fine this way too.